src/NuGet.CommandLine.StagingExtensions/StageCommand.cs

```diff
@@ -146,6 +179,144 @@ private bool TryParseArguments(out SubCommand subCommand, out string parameter)
+
+            Console.PrintJustified(1, string.Format(CultureInfo.CurrentCulture, "{0}:", StagingResources.HeaderStagePackages));
+
+            if (stageDetails.PackagesCount > 0)
+            {
+                foreach (var package in stageDetails.Packages)
+                {
+                    Console.PrintJustified(2, string.Format(CultureInfo.CurrentCulture, "{0}, {1}", package.Id, package.Version));
+                }
+            }
+            else
+            {
+                Console.PrintJustified(2, string.Format(CultureInfo.CurrentCulture, "{0}", StagingResources.None));
+            }
+        }
+
+        private async Task GetCommitProgress(StageManagementResource stageManagementResource, string stageId)
+        {
+            const int checkIntervalSeconds = 10;
+            bool done = false;
+
+            while (!done)
+            {
+                var progress = await stageManagementResource.GetCommitProgress(stageId, Console);
+
+                if (progress.CommitStatus == CommitPending)
+                {
+                    Console.PrintJustified(0, string.Format(CultureInfo.CurrentCulture, "{0}.", StagingResources.StageCommitPending));
+                }
+                else if (progress.CommitStatus == CommitInProgress)
+                {
+                    Console.PrintJustified(0, string.Format(CultureInfo.CurrentCulture, "{0}:", StagingResources.StageCommitInProgress));
+
+                    PrintPackagesProgress(progress);
+
+                    Console.WriteLine();
+                }
+                else if (progress.CommitStatus == CommitCompleted)
+                {
+                    Console.PrintJustified(0, string.Format(CultureInfo.CurrentCulture, "{0}.", StagingResources.StageCommitCompleted));
+
+                    done = true;
+                }
+                else if (progress.CommitStatus == CommitFailed)
+                {
+                    Console.WriteError(string.Format(CultureInfo.CurrentCulture, "{0}. Error: {1}", StagingResources.StageCommitFailed, progress.ErrorMessage));
+
+                    PrintPackagesProgress(progress);
+
+                    Console.WriteWarning(string.Format(CultureInfo.CurrentCulture, "{0}.", StagingResources.RetryCommitMessage));
+
+                    done = true;
+                }
+
+                if (!done)
+                {
+                    for (int i = 0; i < checkIntervalSeconds; i++)
+                    {
+                        await Task.Delay(TimeSpan.FromSeconds(1));
+                        Console.Write(".");
+                    }
+
+                    Console.WriteLine();
+                }
+            }
+        }
+
+        private async Task CommitStage(StageManagementResource stageManagementResource, string apiKey, string stageId)
+        {
+            if (!NonInteractive)
+            {
+                if (!Console.Confirm(string.Format(CultureInfo.CurrentCulture, StagingResources.ConfirmStageCommit, stageId)))

```

Comment:
IMO it's more readable this way

---
In reply to: [78217913](https://github.com/NuGet/NuGet.Staging/pull/73#discussion_r78217913) [](ancestors = 78217913)